internal/pkg/versioncheck/state.go

```diff
@@ -0,0 +1,55 @@
+package versioncheck
+
+import (
+	"encoding/json"
+	"fmt"
+	"os"
+	"time"
+
+	"github.com/hashicorp/hcp/internal/pkg/api/releasesapi/models"
+)
+
+// versionCheckState captures when the last check for a new version of the CLI
+// occurred.
+type versionCheckState struct {
+	CheckedAt time.Time `json:"checked_at"`
+
+	// latestRelease is the latest version of the CLI.
+	latestRelease *models.ProductReleaseResponseV1
```

Comment:
Should we also store the result of the last check so we can keep displaying the message even if we do not perform another check?

Reply:
The reason to not prompt on every run wasn't to reduce load on releases but rather to not annoy the operator. Make sense?